enumerables.rb

```diff
@@ -0,0 +1,81 @@
+module Enumerable
+  def my_each
+    length.times do |v|
+      yield(self[v])
+    end
+  end
+
+  def my_each_with_index
+    length.times do |v|
+      yield(self[v], v)
+    end
+  end
+
+  def my_select
+    arr = []
+    length.times do |v|
+      arr.push(self[v]) if yield(self[v])
+    end
+    arr
+  end
+
+  def my_all?
+    result = true
+    length.times do |v|
+      result = false if yield(self[v]) != true
+      break unless result == true
+    end
+    result
+  end
+
+  def my_any?
+    result = false
+    length.times do |v|
+      result = true if yield(self[v])
+      break unless result == false
+    end
+    result
+  end
+
+  def my_none?
+    result = true
+    length.times do |v|
+      result = false if yield(self[v])
+      break unless result == true
+    end
+    result
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
array = Array.new(100) { rand(0...9) }
array.my_none?(String) == true #true
```
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
words = %w[dog door rod blade]
words.my_none?(5) == words.none?(5) #true
```
- [ ] Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
 words.my_none?(/z/) == words.none?(/z/) #true
 words.my_none?(/d/) == words.none?(/d/) #true
```
- [ ] when no block or argument is given
```ruby
true_array = [nil, false, true, []]
false_array = [nil, false, nil, false]
false_array.my_none? #true
true_array.my_none? #false
```